src/components/DataDimension/ItemOptionsSelector/ItemOptionsSelector.js

```diff
@@ -0,0 +1,302 @@
+                type: item.dimensionItemType,
+                expression: item.expression,
+                optionSetId: item.optionSetId,
+            })
+        })
+        setState((state) => ({
+            ...state,
+            loading: false,
+            options: page > 1 ? [...state.options, ...newOptions] : newOptions,
+            nextPage: result.nextPage,
+        }))
+        /*  The following handles a very specific edge-case where the user can select all items from a
+            page and then reopen the modal. Usually Transfer triggers the onEndReached when the end of
+            the page is reached (scrolling down) or if too few items are on the left side (e.g. selecting
+            49 items from page 1, leaving only 1 item on the left side). However, due to the way Transfer
+            works, if 0 items are available, more items are fetched, but all items are already selected
+            (leaving 0 items on the left side still), the onReachedEnd won't trigger. Hence the code below:
+            IF there is a next page AND some options were just fetched AND you have the same or more
+            selected items than fetched items AND all fetched items are already selected -> fetch more!
+        */
+        if (
+            result.nextPage &&
+            newOptions.length &&
+            selectedItems.length >= newOptions.length &&
+            newOptions.every((newOption) =>
+                selectedItems.find(
+                    (selectedItem) => selectedItem.value === newOption.value
+                )
+            )
+        ) {
+            fetchItems(result.nextPage)
+        }
+    }
+
+    useEffect(() => {
+        setState((state) => ({
+            ...state,
+            filter: {
+                ...state.filter,
+                ...formatOptionsFilters(dataItemType, dataItemId),
+            },
+        }))
+    }, [dataItemId, dataItemType])
+
+    useDidUpdateEffect(() => {
+        fetchItems(1)
+    }, [debouncedSearchTerm, state.filter?.dataItemId])
+
+    const onChange = (selectedIds) => {
+        const newSelectedItems = selectedIds.map((id) => {
+            const matchingItem = [...state.options, ...selectedItems].find(
+                (item) => item.value === id
+            )
+
+            return {
+                value: id,
+                label: matchingItem.label,
+                type: matchingItem.type,
+                optionSetId: matchingItem.optionSetId,
+                ...(matchingItem.expression
+                    ? { expression: matchingItem.expression }
+                    : {}),
+            }
+        })
+
+        onSelect(newSelectedItems)
+    }
+    const onEndReached = () => {
+        if (state.nextPage) {
+            fetchItems(state.nextPage)
+        }
+    }
+    const isActive = (value) => {
+        const item = selectedItems.find((item) => item.value === value)
+        return !item || item.isActive
```

Comment:
This will return `true` if the item is not found. Intentional?

Reply:
Most of the code here is copied from `ItemSelector`.
I'm not totally sure but I think an item on the source side can be active but is not necessarily in the list of selected items.